contrail-agent/hooks/common_utils.py

```diff
@@ -201,3 +207,88 @@ def render_and_log(template, conf_file, ctx, perms=0o444):
         log("Configuration file has been changed.")
 
     return bool(new_set or old_set)
+
+
+def update_nrpe_config(name):
+    plugins_dir = '/usr/local/lib/nagios/plugins'
+    nrpe_compat = nrpe.NRPE()
+    component_ip = get_ip()
+    _rsync_nrpe_checks(plugins_dir)
+    _add_nagios_to_sudoers()
+
+    if name == 'analytics':
+        check_api_cmd = 'check_http -H {} -p 8081'.format(component_ip)
+        nrpe_compat.add_check(
+            shortname='check_analytics_api',
+            description='Check Contrail Analytics API',
+            check_cmd=check_api_cmd
+        )
+    elif name == 'controller':
+        check_ui_cmd = 'check_http -H {} -p 8143 -S'.format(component_ip)
+        nrpe_compat.add_check(
+            shortname='check_contrail_web_ui',
+            description='Check Contrail WebUI',
+            check_cmd=check_ui_cmd
+        )
+
+        check_api_cmd = 'check_http -H {} -p 8082'.format(component_ip)
+        nrpe_compat.add_check(
+            shortname='check_contrail_api',
+            description='Check Contrail API',
+            check_cmd=check_api_cmd
+        )
+
+    ctl_status_shortname = 'check_contrail_status_{}'.format(name)
+    nrpe_compat.add_check(
+        shortname=ctl_status_shortname,
+        description='Check contrail-status',
+        check_cmd=_contrail_status_cmd(name, plugins_dir)
+    )
+
+    nrpe_compat.write()
+
+
+def _rsync_nrpe_checks(plugins_dir):
```

Comment:
I'm sorry - might be I was not so clear. But I asked just for two functions that are equal for all.
function update_nrpe_config has too much specific to be in common - please move it back.
function _contrail_status_cmd can be common only if will not contain check for 'analytics' and will pass cver to all scripts (script can use or ignore parameter)